Source/WebInspectorUI/UserInterface/Views/ResourceTreeElement.js

```diff
@@ -84,7 +84,19 @@ WI.ResourceTreeElement = class ResourceTreeElement extends WI.SourceCodeTreeElem
         if (comparisonResult !== 0)
             return comparisonResult;
 
-        // Compare by title when the subtitles are the same.
+        // Compare by title without file extensions when the subtitles are the same.
+        function filenameWithoutExtention (resourceTitle) {
+            let extentionPosition = resourceTitle.lastIndexOf("." + WI.fileExtensionForFilename(resourceTitle));
```

Comment:
Interestingly the problem you appear to be trying to avoid by including the WI.fileExtensionForFilename in the lastIndexOf check, which is non-extension endings, doesn't actually work here. `WI.fileExtensionForFilename` will give you the string following the last index of a `.` in the name, unless the `.` is at the end of the string. Given that, it seems like for your purpose here we could just check for the last index of a `.` and get the same result.

Reply:
Addressed in 673a204c276c6ecb5d1c3dc8d0df42c54c1b0b67.